### PR TITLE
test(@toss/utils): Improve groupBy Function Tests (Translation and Case Sensitivity Enhancement)

### PR DESCRIPTION
**Pull request by JayeHa**
## Overview

1. Translate groupBy test descriptions from Korean to English for clarity and global team accessibility
2. Enhance groupBy tests with case sensitivity check
I added a new test case because the existing code's test cases were only in Korean, and there was no example case to check if the return value of createKey distinguishes between uppercase and lowercase when it's in English

## PR Checklist

- [x] I read and included theses actions below

1. I have read the [Contributing Guide](https://github.com/toss/slash/blob/main/.github/CONTRIBUTING.md)
4. I have written documents and tests, if needed.

### CODE REVIEW

File: packages/common/utils/src/groupBy.spec.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ describe('groupBy 함수는', () => {
     { groupName: '부엉이', value: 6 },
   ];
 
-  it('createKey의 return value를 이용해서 item을 묶는다', () => {
+  it('groups items using the return value of createKey', () => {
```
Comment (okinawaa) [inline packages/common/utils/src/groupBy.spec.ts:13]: I think `it groups` is wrong gramatically
```suggestion
  it('should group items using the return value of createKey', () => {
```

you can refer https://github.com/toss/slash/pull/267#discussion_r1248051773

File: packages/common/utils/src/groupBy.spec.ts [hunk 2]
```diff
@@ -25,7 +25,25 @@ describe('groupBy 함수는', () => {
...
+      apple: [
+        { name: 'apple', value: 1 },
+        { name: 'apple', value: 2 },
+      ],
+      Apple: [{ name: 'Apple', value: 3 }],
+      banana: [{ name: 'banana', value: 4 }],
+    });
+  });
+
+  it('returns an empty object when given an empty array', () => {
```
Comment (okinawaa) [inline packages/common/utils/src/groupBy.spec.ts:46]: same reason with above comment 😄

```suggestion
  it('should return an empty object when given an empty array', () => {
```

**Review by okinawaa — CHANGES_REQUESTED** [review]

File: packages/common/utils/src/groupBy.spec.ts (hunk 1 above)
Comment (JayeHa) [reply packages/common/utils/src/groupBy.spec.ts]: Thank you for letting me know 👍

[chore: correct the test descriptions grammatically.](https://github.com/toss/slash/pull/360/commits/b6d7a8ce28e88d3194eef881977d3942911b3945)

File: packages/common/utils/src/groupBy.spec.ts (hunk 2 above)
Comment (JayeHa) [reply packages/common/utils/src/groupBy.spec.ts]: I modified it as you told me. thanks :)

[chore: correct the test descriptions grammatically.](https://github.com/toss/slash/pull/360/commits/b6d7a8ce28e88d3194eef881977d3942911b3945)

File: packages/common/utils/src/groupBy.spec.ts [hunk 3]
```diff
@@ -25,7 +25,25 @@ describe('groupBy 함수는', () => {
     });
   });
 
-  it('빈 Array를 넣었을 땐 빈 object를 return한다.', () => {
+  it('should group items distinguishing between uppercase and lowercase', () => {
```
Comment (okinawaa) [inline packages/common/utils/src/groupBy.spec.ts:28]: it is good

**Review by okinawaa — APPROVED** [review]

**Pull request merged**